src/senselab/utils/tasks/dimensionality_reduction.py

```diff
@@ -0,0 +1,71 @@
+"""This module provides the implementation of dimensionality reduction."""
+
+import pydra
+import torch
+import umap
+from sklearn.decomposition import PCA as sklearn_PCA
+from sklearn.manifold import TSNE as sklearn_TSNE
+
+
+def compute_dimensionality_reduction(
+    data: torch.Tensor, model: str = "pca", n_components: int = 2, **kwargs: object
+) -> torch.Tensor:
+    """Reduce the dimensionality of the given data using the specified model.
+
+    This function provides three methods: PCA, t-SNE, or UMAP. The reduced data is returned as a PyTorch tensor.
+
+    Args:
+        data (torch.Tensor): Input data tensor of shape (n_samples, n_features).
+        model (str, optional): The dimensionality reduction model to use.
+            Choices are:
+                * "pca" for Principal Component Analysis,
+                * "tsne" for t-Distributed Stochastic Neighbor Embedding
+                * "umap" for Uniform Manifold Approximation and Projection
+        n_components (int, optional): Number of dimensions in the output.
+            Must be less than or equal to the number of features in the input data.
+            Defaults to 2.
+        **kwargs: Additional keyword arguments to pass to the chosen model.
+
+    Returns:
+        torch.Tensor: The reduced data tensor of shape (n_samples, n_components).
+
+    Raises:
+        ValueError: If an invalid model choice is provided or if n_components is
+            greater than the number of features in the input data or less than 0.
+
+    Examples:
+        >>> data = torch.randn(100, 10)  # 100 samples, 10 features
+        >>> reduced_data = compute_dimensionality_reduction(data, model="pca", n_components=2)
```

Comment:
Can you add some more examples including the important arguments for each model?